pkg/apis/meta/v1/deployment/deployments.go

```diff
@@ -0,0 +1,51 @@
+package deployment
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+//HaveDeploymentsInNamespace returns an error if all provided deployment are not installed in the given namespace
+//client: the client to use
+//namespace: The namespace to search in
+//expectedDeploymentNames: The deployment names to search
+func HaveDeploymentsInNamespace(client kubernetes.Interface, namespace string, expectedDeploymentNames []string) error {
+	versionInfo, err := client.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+	klog.V(1).Infof("Server version info: %v", versionInfo)
+
+	deployments := client.AppsV1().Deployments(namespace)
+
+	for _, deploymentName := range expectedDeploymentNames {
+		klog.V(1).Infof("Check if deployment %s exists", deploymentName)
+		deployment, err := deployments.Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if err != nil {
+			klog.V(1).Infof("Error while retrieving deployment %s: %s", deploymentName, err.Error())
+			return err
+		}
+		if deployment.Status.Replicas != deployment.Status.ReadyReplicas {
+			err = fmt.Errorf("Expect %d but got %d Ready replicas", deployment.Status.Replicas, deployment.Status.ReadyReplicas)
+			klog.Errorln(err)
+			return err
+		}
+		for _, condition := range deployment.Status.Conditions {
+			if condition.Reason == "MinimumReplicasAvailable" {
+				if condition.Status != corev1.ConditionTrue {
+					err = fmt.Errorf("Expect %s but got %s", condition.Status, corev1.ConditionTrue)
+					klog.Errorln(err)
+					return err
+				}
+			}
+		}
```

Comment:
feel need more description on comment as it's not just check has those deployments, it also checks replicas

Reply:
Added comments